### ISSUE TITLE
Expression NotHaveDependencyOutsideNamespace(..) detect false positives

### ISSUE THREAD

**Issue by ricfio** [phparkitect/arkitect#179]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Library Version | 0.2.2
| PHP version     | 8.0.11

#### Summary

Expression NotHaveDependencyOutsideNamespace(...) detect false positives in these cases:
- calling a static method using self keyword
- throwing a new Exception

#### Current behavior

```
App\Domain\Assert violates rules
  should not depend on classes outside in namespace App\Domain (on line 14)
  should not depend on classes outside in namespace App\Domain (on line 21)
```

#### How to reproduce

Using this configuration (phparkitect.php):

```
<?php
declare(strict_types=1);

use Arkitect\ClassSet;
use Arkitect\CLI\Config;
use Arkitect\Expression\ForClasses\NotHaveDependencyOutsideNamespace;
use Arkitect\Expression\ForClasses\ResideInOneOfTheseNamespaces;
use Arkitect\Rules\Rule;

return static function (Config $config): void {
    $allClasses = ClassSet::fromDir(__DIR__.'/src');

    $rule_1 = Rule::allClasses()
        ->that(new ResideInOneOfTheseNamespaces('App\Domain'))
        ->should(new NotHaveDependencyOutsideNamespace('App\Domain'))
        ->because('we want protect our domain');

    $config
        ->add($allClasses, ...[$rule_1]);
};
```

to check this class (Assert.php):

```
<?php

declare(strict_types=1);

namespace App\Domain;

use InvalidArgumentException;

final class Assert
{
    public static function arrayOf(string $class, array $items): void
    {
        foreach ($items as $item) {
            self::instanceOf($class, $item);
        }
    }

    public static function instanceOf(string $class, object $item): void
    {
        if (!$item instanceof $class) {
            throw new InvalidArgumentException(
                sprintf('The object <%s> is not an instance of <%s>', $class, get_class($item))
            );
        }
    }
}

```
#### Expected behavior

No Errors

**Comment by ricfio** [comment]
The first false positive (on line 14) not happens replacing 'self' keyword with ClassName Assert:
```
    public static function arrayOf(string $class, array $items): void
    {
        foreach ($items as $item) {
            Assert::instanceOf($class, $item);
        }
    }
```

**Comment by AlessandroMinoccheri** [comment]
Fixed by #197
